File: android/src/main/kotlin/com/appodeal/appodeal_flutter/AppodealAdViewFactory.kt
package com.appodeal.appodeal_flutter

import android.app.Activity
import android.content.Context
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

internal class AppodealAdViewFactory(
    private val activity: Activity,
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    override fun create(context: Context?, viewId: Int, args: Any?): PlatformView =
        AppodealAdView(activity, args as HashMap<*, *>)
}

internal class AppodealNativeAdViewFactory(
    private val activity: Activity,
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    override fun create(context: Context?, viewId: Int, args: Any?): PlatformView =
        AppodealNativeAdView(activity, args as HashMap<*, *>)
}

File: ios/Classes/AppodealAdViewFactory.swift
import Flutter
import Foundation
import Appodeal

internal class AppodealAdViewFactory: NSObject, FlutterPlatformViewFactory {
    
    private let bannerChannel: FlutterMethodChannel
    private let mrecChannel: FlutterMethodChannel
    
    internal init(mrecChannel: FlutterMethodChannel, bannerChannel: FlutterMethodChannel) {
        self.mrecChannel = mrecChannel
        self.bannerChannel = bannerChannel
    }
    
    func create(withFrame frame: CGRect, viewIdentifier viewId: Int64, arguments args: Any?) -> FlutterPlatformView {
        AppodealAdView(
            frame: frame,
            viewId: viewId,
            args: args as? [String: Any] ?? [:],
            mrecChannel: mrecChannel,
            bannerChannel: bannerChannel
        )
    }
    
    func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
        FlutterStandardMessageCodec.sharedInstance()
    }
}

internal class AppodealNativeAdViewFactory: NSObject, FlutterPlatformViewFactory {
    
    func create(withFrame frame: CGRect, viewIdentifier viewId: Int64, arguments args: Any?) -> FlutterPlatformView {
        AppodealNativeAdView(
            frame: frame,
            viewId: viewId,
            args: args as? [String: Any] ?? [:]
        )
    }
    
    func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
        FlutterStandardMessageCodec.sharedInstance()
    }
}
